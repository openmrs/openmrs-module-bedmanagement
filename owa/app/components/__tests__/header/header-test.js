import React from 'react';
import { StaticRouter } from 'react-router-dom';
import {mount} from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Header from 'components/header';
import UrlHelper from 'utilities/urlHelper';

jest.mock('utilities/urlHelper', () => jest.fn());
UrlHelper.mockImplementation(
    () => ({
        owaPath: () => {
            return '/owa/bedmanagement';
        },
        originPath: () => {
            return '';
        }
    })
);

describe('Header', () => {
    it('renders correctly', () => {
        const header = mount(
            <StaticRouter context={{}}>
                <Header path="/owa/bedmanagement/admissionLocations.html" />
            </StaticRouter>
        );

        expect(header.find('ul').find('li').length).toBe(4);
        expect(header.find('a.active').text().trim()).toBe('Admission Locations');
        expect(shallowToJson(header)).toMatchSnapshot();
    });
});