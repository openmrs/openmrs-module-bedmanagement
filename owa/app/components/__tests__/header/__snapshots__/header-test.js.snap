// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header
      path="/owa/bedmanagement/admissionLocations.html"
    >
      <nav>
        <ul
          className="title-section"
        >
          <li>
            <Link
              className="active"
              replace={false}
              to="/owa/bedmanagement/admissionLocations.html"
            >
              <a
                className="active"
                href="/owa/bedmanagement/admissionLocations.html"
                onClick={[Function]}
              >
                Admission Locations
              </a>
            </Link>
          </li>
          <li>
            <Link
              className=""
              replace={false}
              to="/owa/bedmanagement/bedTypes.html"
            >
              <a
                className=""
                href="/owa/bedmanagement/bedTypes.html"
                onClick={[Function]}
              >
                Bed Types
              </a>
            </Link>
          </li>
          <li>
            <Link
              className=""
              replace={false}
              to="/owa/bedmanagement/bedTags.html"
            >
              <a
                className=""
                href="/owa/bedmanagement/bedTags.html"
                onClick={[Function]}
              >
                Bed Tags
              </a>
            </Link>
          </li>
        </ul>
      </nav>
    </Header>
  </Router>
</StaticRouter>
`;
